File: Math challenge/Challanges/Squares.m
//
//  Squares.m
//  Math challenge
//
//  Created by Lukas Bischof on 09.01.18.
//  Copyright © 2018 Lukas Bischof. All rights reserved.
//

#import "Squares.h"

@implementation Squares

- (Challange *)next
{
    NSInteger rand = randr(11, 20);
    return [[Challange alloc] initWithText:SWF(@"(%ld)^2", (long)rand) andAnswer:(rand * rand)];
}

- (NSString *)challangeDescription {
    return @"2er Potenzen";
}

@end

File: Math challenge/challenges/Squares.swift
//
// Created by Lukas Bischof on 25.10.19.
// Copyright (c) 2019 Lukas Bischof. All rights reserved.
//

import Foundation

class Squares: MathChallengeCategory {
  let challengeDescription: String = "Quadrate"

  func next() -> Challenge {
    let random = randomInt(3, 20)

    return Challenge(text: "\(random)^2", andAnswer: random * random)
  }
}
